src/geometry/PolySetUtils.cc

```diff
@@ -129,4 +141,38 @@ bool is_approximately_convex(const PolySet& ps) {
 #endif
 }
 
+
+PolySet  convert_polyset(const shared_ptr<const Geometry>& geom)
+{
+  PolySet ps(3);
+  if (const auto geomlist = dynamic_pointer_cast<const GeometryList>(geom)) {
+    for (const Geometry::GeometryItem& item : geomlist->getChildren()) {
+      return convert_polyset(item.second);
+    }
+  } else if (const auto N = dynamic_pointer_cast<const CGAL_Nef_polyhedron>(geom)) {
+    bool err = CGALUtils::createPolySetFromNefPolyhedron3(*(N->p3), ps);
+    if (err) {
+      LOG(message_group::Error, "Nef->PolySet failed");
+    } else {
+     return ps;
+    }
+  } else if (const auto ps = dynamic_pointer_cast<const PolySet>(geom)) {
+    return *ps;
+  } else if (const auto hybrid = dynamic_pointer_cast<const CGALHybridPolyhedron>(geom)) {
+    // TODO(ochafik): Implement append_geometry(Surface_mesh) instead of converting to PolySet
+    const PolySet *psp =hybrid->toPolySet().get();
+    return *psp;
+#ifdef ENABLE_MANIFOLD
+  } else if (const auto mani = dynamic_pointer_cast<const ManifoldGeometry>(geom)) {
+    const PolySet *psp = mani->toPolySet().get();
+    return *psp;
+#endif
+  } else if (dynamic_pointer_cast<const Polygon2d>(geom)) { // NOLINT(bugprone-branch-clone)
+    assert(false && "Unsupported file format");
```

Comment:
This error message is a bit misleading; the real issue is that PolySet doesn't support 2D geometry, right?